inference-engine/src/mkldnn_plugin/nodes/mkldnn_one_hot_node.cpp

```diff
@@ -56,32 +49,62 @@ MKLDNNOneHotNode::MKLDNNOneHotNode(const std::shared_ptr<ngraph::Node>& op, cons
-    const auto offValueNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(oneHot->get_input_node_shared_ptr(OFF_VALUEAXES_ID));
-    depth = depthNode->cast_vector<uint32_t>()[0];
+    if (depthNode) {
+        depth = depthNode->cast_vector<uint32_t>()[0];
+    }
     axis = oneHot->get_axis();
-    src_dims = oneHot->get_input_shape(INDICES_ID);
-    if (ngraph::is_scalar(src_dims)) {
-        src_dims = SizeVector{1};
+
+    VectorDims srcDims = getInputShapeAtPort(INDICES_ID).getDims();
+    if (ngraph::is_scalar(srcDims)) {
+        srcDims = SizeVector{1};
     }
-    dst_dims = oneHot->get_output_shape(0);
-    if (ngraph::is_scalar(dst_dims)) {
-        dst_dims = SizeVector{1};
+    VectorDims dstDims = getOutputShapeAtPort(0).getDims();
+    if (ngraph::is_scalar(dstDims)) {
+        dstDims = SizeVector{1};
     }
 
-    int output_dims_size = dst_dims.size();
+    int output_dims_size = dstDims.size();
     if (axis < 0) {
         axis += output_dims_size;
     }
     if (axis < 0 || axis >= output_dims_size) {
         IE_THROW() << errorPrefix << " has unsupported 'axis' attribute: " << oneHot->get_axis();
     }
 
-    if (!( ((1 + src_dims.size()) == dst_dims.size()) ||
-           (src_dims.size() == 1 && dst_dims.size() == 1 && dst_dims[0] == depth && src_dims[0] == 1)))
+    if (!(((1 + srcDims.size()) == dstDims.size()) ||
+            (depthNode && (srcDims.size() == 1 && dstDims.size() == 1 && dstDims[0] == depth && srcDims[0] == 1))))
         IE_THROW() << errorPrefix << " has incorrect number of input/output dimensions!";
 }
 
+bool MKLDNNOneHotNode::needShapeInfer() const {
+    const auto depthNodePtr = reinterpret_cast<int32_t *>(getParentEdgesAtPort(1)[0]->getMemoryPtr()->GetPtr());
+    if (depth != depthNodePtr[0])
+        return true;
+    return MKLDNNNode::needShapeInfer();
+}
+
+std::vector<VectorDims> MKLDNNOneHotNode::shapeInfer() const {
+    std::vector<ov::StaticShape> input_shapes = {
+            getParentEdgesAtPort(0)[0]->getMemory().GetShape().getStaticDims(),
+            getParentEdgesAtPort(1)[0]->getMemory().GetShape().getStaticDims(),
+            getParentEdgesAtPort(2)[0]->getMemory().GetShape().getStaticDims(),
+            getParentEdgesAtPort(3)[0]->getMemory().GetShape().getStaticDims()
+    };
+    std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> input_values = {
+            {1, std::make_shared<ngraph::runtime::HostTensor>(ngraph::element::Type_t::i32, VectorDims{ }, getParentEdgesAtPort(1)[0]->getMemory().GetPtr())},
+            {2, std::make_shared<ngraph::runtime::HostTensor>(ngraph::element::Type_t::f32, VectorDims{ }, getParentEdgesAtPort(2)[0]->getMemory().GetPtr())},
+            {3, std::make_shared<ngraph::runtime::HostTensor>(ngraph::element::Type_t::f32, VectorDims{ }, getParentEdgesAtPort(3)[0]->getMemory().GetPtr())}
```

Comment:
These two input values are unnecessary for shape inference, so you can skip making these HostTensors cuz it takes time for nothing.